### ISSUE TITLE
[Feature Requests] Have real support for HTML admin messages

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#97]
**Reported by lwc on 15 Apr 2008 15:12**

**Description:**
HTML admin messages are currently just sent as plaintext messages with a HTML header.

However, UTF-8 encoding doesn't work unless it's really used in a HTML message. As result, admins of UTF-8 sites (the default in Geeklog) have their HTML messages sent as Gibberish.

Please apply my patch, which is a solution at least for UTF-8 real HTML messages.
Additional Information  public_html/lib-common.html's function COM_mail:

``` PHP
    if( $html )
    {
// custom code - start
if ($charset == "utf-8") {
$boundary = md5(uniqid(time()));
        $headers['Content-Type'] = 'multipart/alternative;' . "\r\n\t" . 'boundary="' . $boundary . '"';
$message_type = array(
'plain' => strip_tags($message),
'html' => '<meta http-equiv="Content-Type" content="text/html; charset=' .$charset .'">' .
"\r\n\r\n" . $message
);
$search = array('format', 'message');
$message_part = '--' . $boundary . '
Content-Type: text/' . $search[0] . '; charset=' . $charset . "\r\n\r\n" . $search[1] . "\r\n\r\n";
$message_temp = '';
foreach ($message_type as $key => $value) {
$replace = array($key, $value);
$message_temp .= str_replace($search, $replace, $message_part);
}
$message_temp .= '--' . $boundary . '--';
$message = $message_temp;
} else
// custom code - end
        $headers['Content-Type'] = 'text/html; charset=' . $charset;
        $headers['Content-Transfer-Encoding'] = '8bit';
    }
```

[Mantis Bugtracker #605](http://project.geeklog.net/tracking/view.php?id=605)

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 15 Apr 2008 18:48**

In other words, you&#039;re asking for proper MIME encoding of HTML mails, RFC 1521 and all that ...

**Comment by mystralkk** [comment]
**Comment posted by lwc on 9 May 2008 17:19**

If you say so. I think my patch is proper enough.

**Comment by mystralkk** [comment]
**Comment posted by THEMike on 14 Sep 2008 09:21**

Will merge in post release of 1.5.1

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 14 Sep 2008 09:48**

The above code should be checked for compliance with RFC 1521 first, though.

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 10 Apr 2009 21:43**

Low priority - pushing it back into the Future.

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 7 Nov 2009 12:31**

On a side note: The non-HTML part of the email should probably be better formatted than simply using the result of a strip_tags().

We now have COM_getTextContent() but that is NOT the correct function to use here. A function that translates HTML to plain text (and does something clever with BR tags and HTML paragraphs, for example) would be useful in other places, too.

**Comment by mystralkk** [comment]
Implemented with change set 4fce878.
